### ISSUE TITLE
[Bug]: diagonal lines in pcolormesh with Gouraud shading and transparency

### ISSUE THREAD

**Issue by limwz01** [matplotlib/matplotlib#27016]
### Bug summary

Diagonal and box lines are produced by `pcolormesh` using a cmap with transparency.

### Code for reproduction

```python
import matplotlib.colors as m_colors
import matplotlib.pyplot as plt
import numpy as np
plt.pcolormesh(np.array([[0, 1, 2], [0, 1, 2]]),
               np.array([[0, 0, 0], [1, 1, 1]]),
               np.array([[0, 1, 2], [0, 1, 2]]),
               cmap=m_colors.LinearSegmentedColormap.from_list(
                   "", [(0, 0, 0, 0), (0, 0, 0, 1)]),
               shading="gouraud")
plt.plot()
```

### Actual outcome

![Figure_1](https://github.com/matplotlib/matplotlib/assets/117669574/714801d4-035a-4869-aab5-95f24e309034)

### Expected outcome

No diagonal or box lines should appear.

### Matplotlib Version

3.6.2

### Matplotlib Backend

QtAgg

### Installation

None

**Comment by AgilentGCMS** [comment]
Just wondering if there's any activity on this one. I routinely plot data on a map using `imshow` in `cartopy`, and would like to use a transparency-aware colormap to show a map background over regions of "uninteresting" data. This is pretty standard in earth science visualization, and perhaps in other fields too. However, this bug prevents me from doing that.

**Comment by jklymak** [comment]
Changing the title, as this is only an issue with Gouraud shading...

**Comment by AgilentGCMS** [comment]
Yes, I can confirm that this is only a problem for `gouraud` shading. Unfortunately, other shading options don't work for me unless there's an option to reduce edge width to zero (I tried with `edgecolors=None`, doesn't do what I want), so I'm stuck with `gouraud`.

**Comment by AgilentGCMS** [comment]
Indeed, `edgecolors='none'` works fine for plots with just `matplotlib` on a flat canvas. The issue is when using `pcolormesh` in `cartopy`, `edgecolors='none'` and `shading='flat'` still draws visible edges. Or perhaps the edges are only visible on non-rectangular projections. I'll open a `cartopy` ticket for that.

**Comment by AgilentGCMS** [comment]
FYI, opened (this issue)[https://github.com/SciTools/cartopy/issues/2295] with `cartopy`.
